test/module/libs/cache/single_pointer_cache_test.cpp

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <thread>
+
+#include <gtest/gtest.h>
+
+#include "cache/single_pointer_cache.hpp"
+#include "module/shared_model/builders/protobuf/test_block_builder.hpp"
+
+using namespace iroha::cache;
+
+class ConsensusCacheBlockTest : public ::testing::Test {
+  using ConsensusBlockCacheType =
+      SinglePointerCache<shared_model::interface::BlockVariant>;
+
+ protected:
+  void SetUp() override {
+    consensus_cache_block = std::make_shared<ConsensusBlockCacheType>();
+  }
+
+  void TearDown() override {
+    consensus_cache_block.reset();
+  }
+
+  std::shared_ptr<ConsensusBlockCacheType> consensus_cache_block;
```

Comment:
consensus_block_cache?

Reply:
Also why is it pointer?

Reply:
It's a pointer, so it could easily be reseted in ```TearDown()```